source/extensions/quic_listeners/quiche/envoy_quic_alarm.cc

```diff
@@ -11,8 +11,13 @@ EnvoyQuicAlarm::EnvoyQuicAlarm(Event::Dispatcher& dispatcher, const quic::QuicCl
 void EnvoyQuicAlarm::CancelImpl() { timer_->disableTimer(); }
 
 void EnvoyQuicAlarm::SetImpl() {
-  // TODO(#7170) switch to use microseconds if it is supported.
-  timer_->enableTimer(std::chrono::milliseconds(getDurationBeforeDeadline().ToMilliseconds()));
+  quic::QuicTime::Delta duration = getDurationBeforeDeadline();
+  // Round up the duration so that any duration < 1us will not be triggered within current event
+  // loop. QUICHE alarm is not expected to be scheduled in current event loop. This bit is a bummer
+  // in QUICHE, and we are working on the fix. Once QUICHE is fixed of expecting this behavior, we
+  // no longer need to round up the duration.
+  timer_->enableHRTimer(std::chrono::microseconds(
+      (duration < quic::QuicTime::Delta::FromMicroseconds(1)) ? 1 : duration.ToMicroseconds()));
```

Comment:
nit: How about `std::max(1, duration.ToMicroseconds());`

Reply:
Doesn't show up - did you miss an upload?
Also  cc @antoniovicente for if there's a better way to do this, or if it requires your libevent workarounds

Reply:
Please add a TODO(antoniovicente) here.  I'm looking into some improvements here.